packages/compass-query-bar/src/utils/query-storage.spec.ts

```diff
@@ -96,10 +106,52 @@ describe('QueryStorage', function () {
   });
 
   it('deletes file from storage', async function () {
+    queries.forEach((query) => writeQuery(tmpDir, query));
+
     await queryHistoryStorage.delete(queries[0]._id);
     expect(await queryHistoryStorage.loadAll()).to.have.lengthOf(1);
 
     await queryHistoryStorage.delete(queries[1]._id);
     expect(await queryHistoryStorage.loadAll()).to.have.lengthOf(0);
   });
+
+  context(
+    `limits saved recent queries to ${maxAllowedRecentQueries}`,
+    function () {
+      it('truncates recents queries to max allowed queries', async function () {
+        const queries = createNumberOfQueries(tmpDir, maxAllowedRecentQueries);
```

Comment:
What do you think on doing 

```suggestion
        const queries = createNumberOfQueries(tmpDir, maxAllowedRecentQueries + 1);
```

If we use the maximum here, if I understand well the behaviour, we won't test the use case where we surpass the limit.

Reply:
Here I create the max queries and then after this I create another one using `queryHistoryStorage.saveQuery`. This method handles the max connections by removing any exceeding connection. If you check the assertions after this line, it asserts that max connections is same as `maxAllowedRecentQueries`